clean the duplicate

Zhaozhi233 opened the pull request:
The reason of why we have the duplicate problem in most cases is because the name we use in 'param' is different with the name in the function.
For example, here we can change the 'product-id' in the @api.param into 'product_id'(change the '-' into '_'). Then the duplicate parameters will be gone magically.

Zhaozhi233 committed:
clean the duplicate

Zhaozhi233 committed:
merge from master

Zhaozhi233 committed:
clean up

Mu-xsan commented:
Hi George,

Could you please check those delete endpoints?

Although they don't have duplicates at this time but I don't know whether they have the problems you mentioned above.

For unity, I think they'd better be solved in your way.

Zhaozhi233 committed:
bug fix

Zhaozhi233 commented:
I did two things in this new commit.
## 1.
I checked if we can keep the product_id and related_product_id in the argument model.
I failed and did find a way to keep it. The problem is kind of tricky.
In function, we can only use "\_", however, in the url we suppose to use "-" according to professor's requirement.
So if we keep the product id and related product id in the arguments model, we need use "-". But the duplication will definitely exist because we need use "\_" in the function.
So I think it's the only solution as Muxin did that we delete product id and related product id in the arguments model.
## 2.
I modified the search function a little bit. I make the status type to bool in the @api.param. 
After doing this, a screw down button will show in the api doc GET method which will looks pretty.
However, the input status is still a string format. 
To make it pass the apidoc method, I modified the status check from "by_status == "True"" to "by_status == "True" or by_status == "true"".
